deployment/docker-swarm/database-utilities.sh

```diff
@@ -0,0 +1,149 @@
+#!/bin/bash
+# defaults
+backup=false
+restore=false
+file=
+databaseKeyword=alertdb
+databaseName=alertdb
+userName=sa
+password=
+
+# functions
+usage() {
+  echo "usage: database-utilities - backup or restore a database with docker."
+  echo
+  echo "database-utilities.sh [-b] [-f file] [-d databaseName] [-k containerKeyword] [-r] [-u userName]"
+  echo "Options: "
+  echo "  -b: backup a database to the file specified in the file option."
+  echo "  -f: the file to save a backup or the file to restore the database from."
+  echo "  -d: the name of the database."
+  echo "  -k: the keyword to search for the database container."
+  echo "  -r: restore a database from the file specified by the file option."
+  echo "  -u: database user name."
+  echo "  -h: display this help."
+  echo
+  echo "Examples:"
+  echo "  Backup:"
+  echo "    database-utilities.sh -b -f ~/my-db-backup.dump"
+  echo "  Restore:"
+  echo "    database-utilities.sh -r -f ~/my-db-backup.dump"
+  echo
+  echo
+}
+
+displayConfiguration() {
+  if [ -z $password ];
+    then
+      databasePasswordSet=false
+    else
+      databasePasswordSet=true
+  fi
+  echo "-------------------------------------"
+  echo "Configured Options: "
+  echo "  Mode:"
+  echo "    backup:  $backup"
+  echo "    restore: $restore"
+  echo "  Backup/Restore File: $file"
+  echo "  Container Search Keyword: $databaseKeyword"
+  echo "  Database Name: $databaseName"
+  echo "  Database User: $userName"
+  echo "  Database Password Set: $databasePasswordSet"
+  echo "-------------------------------------"
+}
+
+checkContainerFound() {
+  if [ -z $containerId ];
+    then
+      echo
+      echo "Database container not found exiting ..."
+      echo
+      exit 1
+    else
+      echo
+      echo "Database container found with ID: $containerId"
+      echo
+  fi
+}
+
+checkFileSpecified() {
+  if [ -z $file ];
+    then
+      echo "-f file option not specified. $1"
+      echo
+      exit 1
+  fi
+}
+
+backupDatabase() {
+  checkFileSpecified "Cannot backup the database."
+  echo "Backing up database $databaseName"
+  docker exec -i $containerId pg_dump -Fc -U $userName -f /tmp/alert-database.dump $databaseName;
+  docker cp $containerId:/tmp/alert-database.dump $file
```

Comment:
Since you aren't checking the status of "docker exec", if it fails I believe the script will continue on to "docker cp", and then the echo statement with success.
You could add the builtin "set -e", which will fail the script if any command fails.